plugins/catalog-backend/src/next/database/DefaultProcessingDatabase.ts

```diff
@@ -508,6 +518,61 @@ export class DefaultProcessingDatabase implements ProcessingDatabase {
     };
   }
 
+  async listAncestors(
+    txOpaque: Transaction,
+    options: ListAncestorsOptions,
+  ): Promise<ListAncestorsResult> {
+    const tx = txOpaque as Knex.Transaction;
+    const { entityRef } = options;
+    const entityRefs = new Array<string>();
+
+    let currentRef = entityRef;
+    let depth = 0;
+    for (;;) {
+      if (depth++ > MAX_ANCESTOR_DEPTH) {
+        throw new Error(
+          `Unable receive ancestors for ${entityRef}, reached maximum depth of ${MAX_ANCESTOR_DEPTH}`,
+        );
+      }
+
+      const rows = await tx<DbRefreshStateReferencesRow>(
+        'refresh_state_references',
+      )
+        .where({ target_entity_ref: currentRef })
+        .select();
+
+      if (rows.length === 0) {
+        if (depth === 1) {
+          throw new NotFoundError(`Entity ${currentRef} not found`);
+        }
+        throw new NotFoundError(
+          `Entity ${entityRef} has a broken parent reference chain at ${currentRef}`,
+        );
+      }
+
+      const parentRef = rows.find(r => r.source_entity_ref)?.source_entity_ref;
+      if (!parentRef) {
+        // We've reached the top of the tree which is the entityProvider.
+        // In this case we refresh the entity itself.
```

Comment:
is this comment correct? it seems we actually return the reference chain found so far, and i'm a bit unclear how this comment applies here

Reply:
Yeah this is just the exit now, it used to be the point where we did the refresh before splitting apart refresh and ancestry discovery

Reply:
So still should remove this row